### PR TITLE
Allow decoding using epochs at different stages of pre-processing

### LINKED ISSUE #808: Add on option to use "un-cleaned" epochs for decoding (omitting ICA / SSP)
My sense is that things like SSP and ICA probably won't matter but that peak-to-peak rejection could help, depending on the nature of the artifacts. That sounds in line with what you're seeing @hoechenberger . In this case it makes sense for us to use the "clean" epochs for decoding. At some point we could make it an option which to use (but maybe assume YAGNI for now?)

_Originally posted by @larsoner in https://github.com/mne-tools/mne-bids-pipeline/issues/796#issuecomment-1789318605_

### PR DESCRIPTION
**Pull request by hoechenberger**
Closes #808

- [x] Changelog has been updated (`docs/source/changes.md`)

### CODE REVIEW

File: docs/source/v1.5.md.inc [hunk 1]
```diff
@@ -22,6 +22,7 @@ All users are encouraged to update.
 - Added support for "local" [`autoreject`](https://autoreject.github.io) to remove bad epochs
   before submitting the data to ICA fitting. This can be enabled by setting [`ica_reject`][mne_bids_pipeline._config.ica_reject]
   to `"autoreject_local"`. (#810, #816 by @hoechenberger)
+- The new setting [`decoding_which_epochs`][mne_bids_pipeline._config.decoding_which_epochs] controls which epochs (e.g., uncleaned, after ICA/SSP, cleaned) shall be used for decoding. (#xxx by @hoechenber)
```
Comment (hoechenberger) [inline docs/source/v1.5.md.inc:25]:
```suggestion
- The new setting [`decoding_which_epochs`][mne_bids_pipeline._config.decoding_which_epochs] controls which epochs (e.g., uncleaned, after ICA/SSP, cleaned) shall be used for decoding. (#819 by @hoechenber)
```

File: mne_bids_pipeline/_config.py [hunk 2]
```diff
@@ -1465,10 +1465,41 @@
...
+decoding_which_epochs: Literal[
+    "uncleaned", "after_ica", "after_ssp", "cleaned"
+] = "cleaned"
+"""
+This setting controls which epochs will be fed into the decoding algorithms.
+
+!!! info
+    Decoding is a very powerful tool that often can deal with noisy data surprisingly
+    well. Depending on the specific type of data, artifacts, and analysis performed,
+    decoding performance may even improve with less "pre-processed" the data, as
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:1477]:
```suggestion
    decoding performance may even improve with less pre-processed data, as
```

**Comment by hoechenberger** [comment]
Rendered docs:
https://output.circle-artifacts.com/output/job/385027e0-63d8-4b1a-b972-9531f649039e/artifacts/0/site/settings/sensor/mvpa.html#mne_bids_pipeline._config.decoding_which_epochs

File: mne_bids_pipeline/_config.py [hunk 3]
```diff
@@ -1465,10 +1465,41 @@
...
+This setting controls which epochs will be fed into the decoding algorithms.
+
+!!! info
+    Decoding is a very powerful tool that often can deal with noisy data surprisingly
+    well. Depending on the specific type of data, artifacts, and analysis performed,
+    decoding performance may even improve with less pre-processed data, as
+    processing steps such as ICA or SSP often remove parts of the signal, too, in
+    addition to noise. By default, MNE-BIDS-Pipeline uses cleaned epochs for decoding,
+    but you may choose to use entirely uncleaned epochs, or epochs before the final
+    PTP-based rejection step.
```
Comment (hoechenberger) [inline mne_bids_pipeline/_config.py:1481]:
```suggestion
    PTP-based rejection or Autoreject step.
```

**Comment by larsoner** [comment]
Nice @hoechenberger !

**Pull request merged**